### PR TITLE
top-level/release.nix: make Darwin builds do more

### PR DESCRIPTION
**Pull request by copumpkin**
@edolstra is this doing what I think it is?

My intent is for `nixpkgs-unstable` to block if `git`, `mysql`, and `vim` don't build, and to ask Hydra to also try to build several other packages (all of which I verified on my own machine to work on Darwin last night).

I also added a blocking job on the full `stdenv` bootstrap from the bootstrap tools specified in make-bootstrap-tools.nix, to make it easier to notice when changes to them break everything, since we don't update them very often.

### CODE REVIEW

File: pkgs/top-level/release.nix [hunk 1]
```diff
@@ -55,10 +59,24 @@ let
...
 
     } // (mapTestOn ((packagePlatforms pkgs) // rec {
 
+      # TODO: most (but possibly not all) of the jobs specified here are unnecessary now that we have release-lib.nix
+      # traversing all packages and looking at their meta.platform attributes. Someone who's better at this than I am
+      # should go through these and kill the ones that are safe to kill.
+      #
+      # <niksnut> note that all that " = linux" stuff in release.nix is legacy, from before we had meta.platforms
+      # <copumpkin> niksnut: so should I just kill all the obsolete jobs in release.nix?
+      # <niksnut> I don't know if they're all covered
```
Comment (vcunat) [inline pkgs/top-level/release.nix]: Yeah, someone just needs to do the cleaning by checking line by line.

**Comment by vcunat** [comment]
I'm fairly confident this does what you intended.

**Comment by copumpkin** [comment]
@vcunat thanks!

**Comment by vcunat** [comment]
:-) The hard part will be actually fixing the builds so the channel isn't blocked by these.

**Comment by copumpkin** [comment]
They already work! We'll make sure that stays true :smile:

**Comment by vcunat** [comment]
We used to have darwin builders for two concurrent jobs only and now there are for 12. That should help, as the speed of feedback is important, I believe.

**Comment by copumpkin** [comment]
I have other evil plans to make that even better. Now I just need an evil plan to get the time needed to work on those other evil plans :smile:

**Comment by vcunat** [comment]
A time machine? (When we're speaking of darwin stuff :)

**Pull request merged**